src/acvp.c

```diff
@@ -4774,6 +4848,163 @@ static ACVP_RESULT acvp_build_dsa_register_cap (JSON_Object *cap_obj, ACVP_CAPS_
+                alg_str = acvp_prereqs_tbl[i].name;
+                json_object_set_string(obj, "algorithm", alg_str);
+                json_object_set_string(obj, ACVP_PREREQ_VAL_STR, pre_req->val);
+                break;
+            }
+        }
+
+        json_array_append_value(prereq_array, val);
+        next_pre_req = prereq_vals->next;
+        prereq_vals = next_pre_req;
+    }
+    return ACVP_SUCCESS;
+}
+
+static ACVP_RESULT acvp_build_kas_ecc_register_cap (JSON_Object *cap_obj, 
+                                                    ACVP_CAPS_LIST *cap_entry) {
+    JSON_Array *temp_arr = NULL;
+    ACVP_RESULT result;
+    ACVP_KAS_ECC_CAP_MODE *kas_ecc_mode;
+    ACVP_KAS_ECC_CAP *kas_ecc_cap;
+    int i;
+    ACVP_PARAM_LIST *current_func;
+    ACVP_PARAM_LIST *current_curve;
+
+    json_object_set_string(cap_obj, "algorithm", acvp_lookup_cipher_name(cap_entry->cipher));
+
+    for (i=1; i<ACVP_KAS_ECC_MAX_MODES; i++) {
+        kas_ecc_cap = cap_entry->cap.kas_ecc_cap;
+        kas_ecc_mode = &kas_ecc_cap->kas_ecc_mode[i-1];
+        if (kas_ecc_mode->prereq_vals) {
+            switch (kas_ecc_mode->cap_mode)
+            {
+            case ACVP_KAS_ECC_MODE_CDH:
+                json_object_set_string(cap_obj, "mode", "CDH-Component");
+                break;
+            case ACVP_KAS_ECC_MODE_NONE:
+            case ACVP_KAS_ECC_MODE_COMPONENT:
+            default:
+                break;
+            }
+            result = acvp_lookup_kas_ecc_prereqVals(cap_obj, kas_ecc_mode);
+            if (result != ACVP_SUCCESS) { return result; }
+        }
+        switch (i)
+        {
+        case ACVP_KAS_ECC_MODE_CDH:
+
+            json_object_set_value(cap_obj, "function", json_value_init_array());
+            temp_arr = json_object_get_array(cap_obj, "function");
+            current_func = kas_ecc_mode->function;
+            while (current_func) {
+                switch (current_func->param)
+                { 
+                case ACVP_KAS_ECC_FUNC_PARTIAL:
+                    json_array_append_string(temp_arr, "partialVal");
+                    break;
+                case ACVP_KAS_ECC_FUNC_DPGEN:
+                case ACVP_KAS_ECC_FUNC_DPVAL:
+                case ACVP_KAS_ECC_FUNC_KEYPAIR:
+                case ACVP_KAS_ECC_FUNC_KEYREGEN:
+                case ACVP_KAS_ECC_FUNC_FULL:
+                default:
+                    printf("\nUnsupported KAS-ECC function %d", current_func->param);
+                    break;
+                }
+                current_func = current_func->next;
+            }
+            json_object_set_value(cap_obj, "curve", json_value_init_array());
+            temp_arr = json_object_get_array(cap_obj, "curve");
+            current_curve = kas_ecc_mode->curve;
+            while (current_curve) {
+                switch (current_curve->param)
+                { 
+                case ACVP_ECDSA_CURVE_B233:
+                    json_array_append_string(temp_arr, "b-233");
```

Comment:
Replace strings with #defines?

Reply:
pass for now - there so many other places. Maybe we could make a user story to move them all to #defines